sensors/sensors_test.go

```diff
@@ -91,56 +93,108 @@ func TestNewLidar(t *testing.T) {
 	t.Run("Successful lidar creation", func(t *testing.T) {
 		lidar := "good_lidar"
 		ctx := context.Background()
-		deps := s.SetupDeps(lidar)
+		deps := s.SetupDeps(lidar, "")
 		actualLidar, err := s.NewLidar(ctx, deps, lidar, logger)
 		test.That(t, actualLidar.Name, test.ShouldEqual, lidar)
 		test.That(t, err, test.ShouldBeNil)
 
-		tsr, err := actualLidar.TimedSensorReading(ctx)
+		tsr, err := actualLidar.TimedLidarSensorReading(ctx)
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, len(tsr.Reading), test.ShouldBeGreaterThan, 0)
 	})
 }
 
-func TestTimedSensorReading(t *testing.T) {
+func TestNewIMU(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+
+	t.Run("No IMU provided", func(t *testing.T) {
+		imu := ""
+		lidar := "good_lidar"
+		deps := s.SetupDeps(lidar, imu)
+		_, err := s.NewIMU(context.Background(), deps, imu, logger)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("Failed IMU creation with non-existing sensor", func(t *testing.T) {
+		lidar := "good_lidar"
+		imu := "gibberish"
+		deps := s.SetupDeps(lidar, imu)
+		actualIMU, err := s.NewIMU(context.Background(), deps, imu, logger)
+		expectedIMU := s.IMU{}
+		test.That(t, actualIMU, test.ShouldResemble, expectedIMU)
+		test.That(t, err, test.ShouldBeError,
```

Comment:
[nit] best practice is to always check error first

Reply:
Note; I see that we originally checked for the lidar state first why is probably why you ordered it this way. Can you also switch the orders of the LiDAR response as well?

Reply:
Will do